src/lib/stateMappings/__tests__/singleIndex-test.ts

```diff
@@ -0,0 +1,215 @@
+      const stateMapping = singleIndexStateMapping('indexName');
+      const actual = stateMapping.stateToRoute({
+        indexName: {
+          query: 'zamboni',
+          refinementList: {
+            color: ['red'],
+          },
+        },
+        anotherIndex: {
+          // @ts-ignore
+          totally: 'ignored',
+          refinementList: {
+            color: ['blue'],
+          },
+        },
+      });
+
+      expect(actual).toEqual({
+        query: 'zamboni',
+        refinementList: {
+          color: ['red'],
+        },
+      });
+    });
+
+    it('empty object if there is no matching index', () => {
+      const stateMapping = singleIndexStateMapping('magicIndex');
+      const actual = stateMapping.stateToRoute({
+        indexName: {
+          query: 'zamboni',
+          refinementList: {
+            color: ['red'],
+          },
+        },
+        anotherIndex: {
+          // @ts-ignore
+          totally: 'ignored',
+          refinementList: {
+            color: ['blue'],
+          },
+        },
+      });
+
+      expect(actual).toEqual({});
+    });
+  });
+
+  describe('routeToState', () => {
+    it('passes normal state through', () => {
+      const stateMapping = singleIndexStateMapping('indexName');
+      const actual = stateMapping.routeToState({
+        query: 'zamboni',
+        refinementList: {
+          color: ['red'],
+        },
+      });
+
+      expect(actual).toEqual({
+        indexName: {
+          query: 'zamboni',
+          refinementList: {
+            color: ['red'],
+          },
+        },
+      });
+    });
+
+    it('removes configure', () => {
+      const stateMapping = singleIndexStateMapping('indexName');
+      const actual = stateMapping.routeToState({
+        query: 'zamboni',
+        refinementList: {
+          color: ['red'],
+        },
+        configure: {
+          advancedSyntax: false,
+        },
+      });
+
+      expect(actual).toEqual({
+        indexName: {
+          query: 'zamboni',
+          refinementList: {
+            color: ['red'],
+          },
+        },
+      });
+    });
+
+    it('passes non-UiState through', () => {
+      const stateMapping = singleIndexStateMapping('indexName');
+      const actual = stateMapping.routeToState({
+        query: 'zamboni',
+        refinementList: {
+          color: ['red'],
+        },
+        // @ts-ignore
+        spy: ['stealing', 'all', 'your', 'searches'],
+      });
+
+      expect(actual).toEqual({
+        indexName: {
+          query: 'zamboni',
+          refinementList: {
+            color: ['red'],
+          },
+          spy: ['stealing', 'all', 'your', 'searches'],
+        },
+      });
+    });
+
+    it('returns wrong data if used with nested state', () => {
+      const stateMapping = singleIndexStateMapping('indexName');
+      const actual = stateMapping.routeToState({
+        // @ts-ignore (we are passing wrong data)
+        indexName: {
```

Comment:
can you tell me what parts of this object are "wrong data"?

Reply:
routeToState expects a single index here, but this is a multi-index state (which doesn't work for "singleIndexStateMapping"